### PR TITLE
[JUJU-4405] Remove mongo from api tests and also limit dependencies

### PR DESCRIPTION
**Pull request by wallyworld**
The api package tests no longer use mongo. The test suites have been converted to use a test api server.

Also, rename old legacy "st" receivers to "c" or "conn".

The api package also had undesirable dependencies on apiserver and state packages. Some like refactoring has fixed this. eg backup metadata now has been moved to core.

## QA steps

run the tests

### CODE REVIEW

File: api/facadeversions_test.go [hunk 1]
```diff
@@ -19,30 +16,6 @@ type facadeVersionSuite struct {
 
 var _ = gc.Suite(&facadeVersionSuite{})
 
-func (s *facadeVersionSuite) TestFacadeVersionsMatchServerVersions(c *gc.C) {
```
Comment (wallyworld) [inline api/facadeversions_test.go:22]: moved this test to apiserver to get rid of the dependency in api

File: apiserver/common/networkingcommon/types.go [hunk 2]
```diff
@@ -263,28 +261,3 @@ func networkAddressesToStateArgs(
 
 	return res
 }
-
-func FanConfigToFanConfigResult(config network.FanConfig) params.FanConfigResult {
```
Comment (wallyworld) [inline apiserver/common/networkingcommon/types.go:267]: moved to rpc/params

File: apiserver/facades/client/backups/backups.go [hunk 3]
```diff
@@ -87,39 +87,3 @@ func NewAPI(backend Backend, authorizer facade.Authorizer, machineTag names.Tag,
 }
 
 var newBackups = backups.NewBackups
-
-// CreateResult updates the result with the information in the
-// metadata value.
-func CreateResult(meta *backups.Metadata, filename string) params.BackupsMetadataResult {
```
Comment (wallyworld) [inline apiserver/facades/client/backups/backups.go:93]: moved to rpc/params

File: core/backups/archive_data_test.go [hunk 4]
```diff
@@ -120,39 +120,13 @@ func (s *archiveDataSuiteV0) TestVersionFound(c *gc.C) {
...
-}
-
 type baseArchiveDataSuite struct {
 	archiveFile *bytes.Buffer
 	data        []byte
 	meta        *backups.Metadata
 }
 
 const (
-	testMetadataV0 = `{` +
```
Comment (wallyworld) [inline core/backups/archive_data_test.go:141]: v0 is dead to us

File: core/backups/metadata.go [hunk 5]
```diff
@@ -0,0 +1,319 @@
+// Copyright 2014 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package backups
```
Comment (wallyworld) [inline core/backups/metadata.go:4]: moved from state/backups

**Review by hpidcock — APPROVED** [review]

**Pull request merged**